### PR TITLE
Make operations on `set` take `AbstractSet[object]`

### PR DESCRIPTION
**Pull request by ilinum**
Previously they were `AbstractSet[Any]`

### CODE REVIEW

File: stdlib/3/builtins.pyi [hunk 1]
```diff
@@ -682,28 +682,28 @@ class set(MutableSet[_T], Generic[_T]):
...
+    def __ge__(self, s: AbstractSet[object]) -> bool: ...
+    def __gt__(self, s: AbstractSet[object]) -> bool: ...
     # TODO more set operations
 
 class frozenset(AbstractSet[_T], Generic[_T]):
     def __init__(self, iterable: Iterable[_T] = ...) -> None: ...
     def copy(self) -> FrozenSet[_T]: ...
-    def difference(self, *s: Iterable[Any]) -> FrozenSet[_T]: ...
-    def intersection(self, *s: Iterable[Any]) -> FrozenSet[_T]: ...
+    def difference(self, *s: Iterable[object]) -> FrozenSet[_T]: ...
```
Comment (JelleZijlstra) [inline stdlib/3/builtins.pyi]: Hm, `__sub__` below (which is implemented the same way in C) takes `AbstractSet[_T]`; I'm not sure why we can't specify `_T` here.

**Comment by JelleZijlstra** [comment]
Generally looks good; I wonder if we should use `_T` instead of `object` for at least some of these.

File: stdlib/3/builtins.pyi (hunk 1 above)
Comment (ilinum) [inline stdlib/3/builtins.pyi]: Hmm, good catch both of these should be the same. 

However, I think we should change the `__sub__` function, in order to allow the following code:
```py
from typing import Set, Union, FrozenSet

s1 = frozenset([1, 2])  # type: FrozenSet[int]
s2 = frozenset([1, '2'])  # type: FrozenSet[Union[str, int]]

# currently, the following line gives you an error but I don't think it should.
s3 = s1 - s2  # type: FrozenSet[int]
```

What do you think?

Comment (JelleZijlstra) [inline stdlib/3/builtins.pyi]: That makes sense from the perspective of minimizing false positives. It seems a shame that mypy won't tell you if you're mixing sets of different types in your operations (which is surely often a bug), but we should generally err on the side of reducing false positives.

Comment (gvanrossum) [inline stdlib/3/builtins.pyi]: I think we should first decide how to do that for `==`; then maybe we'll be able to generalize for other operations on other objects. I think for `==` there's probably already an open issue.

**Pull request merged**